nautobot/extras/datasources/git.py

```diff
@@ -404,6 +407,21 @@ def import_config_context(context_data, repository_record, job_result, logger):
         data = context_data.copy()
         del data["_metadata"]
 
+        if context_metadata["schema"]:
+            if getattr(context_record.schema, "name", None) != context_metadata["schema"]:
+                try:
+                    schema = ConfigContextSchema.objects.get(name=context_metadata["schema"])
+                    context_record.schema = schema
```

Comment:
Almost missed this! This needs to handle the cases where:

1) No "schema" key is defined on the metadata (don't error out, remove any existing schema from the context)
2) Schema key is set to `None` or `""` on the metadata (don't error out, remove any existing schema from the context)